source/server/server.cc

```diff
@@ -412,8 +412,29 @@ void InstanceImpl::initialize(const Options& options,
     server_compilation_settings_stats_->fips_mode_.set(1);
   }
 
-  bootstrap_.mutable_node()->set_hidden_envoy_deprecated_build_version(VersionInfo::version());
-  bootstrap_.mutable_node()->set_user_agent_name("envoy");
+  // Check to see if bootstrap has set user_agent_name.  If yes, use that.  
+  // If not, use "envoy"
+  std::string user_agent_name = bootstrap_.node().user_agent_name();
+  if (user_agent_name == "" ) {
+    user_agent_name = "envoy";
+  }
+  bootstrap_.mutable_node()->set_user_agent_name(user_agent_name);
```

Comment:
I think it would be more readable if you move setting the value into the `if` statement-true part. Since this is the only way it is modified in the proto.

Same for below cases.

Reply:
sure thing, happy to do whatever is most idiomatic here.  I took the original approach to do a few lines of determining what the right value is, then a single line to set it.  I think if I move the setting into the conditional then I need two lines that call the actual setting.  

I'll throw something up and you can tell me if that's what you were thinking.